src/rules/no-unused-class-name.ts

```diff
@@ -0,0 +1,116 @@
+import { createRule } from "../utils"
+import type {
+  SourceLocation,
+  SvelteAttribute,
+  SvelteDirective,
+  SvelteShorthandAttribute,
+  SvelteSpecialDirective,
+  SvelteSpreadAttribute,
+  SvelteStyleDirective,
+} from "svelte-eslint-parser/lib/ast"
+import type { AnyNode } from "postcss"
+import {
+  default as selectorParser,
+  type Node as SelectorNode,
+} from "postcss-selector-parser"
+
+export default createRule("no-unused-class-name", {
+  meta: {
+    docs: {
+      description:
+        "disallow the use of a class in the template without a corresponding style",
+      category: "Best Practices",
+      recommended: false,
+    },
+    schema: [],
+    messages: {},
+    type: "suggestion",
+  },
+  create(context) {
+    const classesUsedInTemplate: Record<string, SourceLocation> = {}
+
+    return {
+      SvelteElement(node) {
+        if (node.kind !== "html") {
+          return
+        }
+        const classes = node.startTag.attributes.flatMap(findClassesInAttribute)
+        for (const className of classes) {
+          classesUsedInTemplate[className] = node.startTag.loc
+        }
+      },
+      "Program:exit"() {
+        const styleContext = context.parserServices.getStyleContext()
+        if (["parse-error", "unknown-lang"].includes(styleContext.status)) {
+          return
+        }
+        const classesUsedInStyle =
+          styleContext.sourceAst != null
+            ? findClassesInPostCSSNode(styleContext.sourceAst)
+            : []
+        for (const className in classesUsedInTemplate) {
+          if (!classesUsedInStyle.includes(className)) {
+            context.report({
+              loc: classesUsedInTemplate[className],
+              message: `Unused class "${className}".`,
+            })
+          }
+        }
+      },
+    }
+  },
+})
+
+/**
+ * Extract all class names used in a HTML element attribute.
+ */
+function findClassesInAttribute(
+  attribute:
+    | SvelteAttribute
+    | SvelteShorthandAttribute
+    | SvelteSpreadAttribute
+    | SvelteDirective
+    | SvelteStyleDirective
+    | SvelteSpecialDirective,
+): string[] {
+  if (attribute.type === "SvelteAttribute" && attribute.key.name === "class") {
+    return attribute.value.flatMap((value) =>
+      value.type === "SvelteLiteral" ? value.value.split(" ") : [],
```

Comment:
I think we should use regex because we can also use tabs etc.

```suggestion
      value.type === "SvelteLiteral" ? value.value.split(/\s+/u) : [],
```